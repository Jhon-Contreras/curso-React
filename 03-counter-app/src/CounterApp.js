
// importamos PropTypes para manejar el props  value 
import PropTypes from 'prop-types';

// el componente CounterApp es un functional component y recibe el props (desestructurado) 
const CounterApp = ( {value} ) =>{

    //handleAdd
    const handleAdd = (e) =>{
        console.log(e);
    } 
    // retornamos el jsx 
    return (
        <>
            <h1> CounterApp </h1>
            <p> { value } </p>
            <button onClick={ handleAdd }>+1</button>
        </>
    );
}

// establecemos que value es de tipo numero y es requerido 
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

// exportamos el componente 
export default CounterApp;