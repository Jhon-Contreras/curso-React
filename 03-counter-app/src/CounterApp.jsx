
// importamos PropTypes para manejar el props  value 
import React, { useState } from 'react'
import PropTypes from 'prop-types';

// el componente CounterApp es un functional component y recibe el props (desestructurado) 
const CounterApp = ( {value = 10} ) =>{

    const [ counter, setCounter ] = useState(value); //retorna un [] con 2 valores

    //handleAdd
    const handleAdd = (e) =>{
        // setCounter( counter + 1)
        // esta es otra alternativa 
        setCounter( (c) => c + 1  )
    } 
    // handleSubtract
    const handleSubtract = (e) =>{
        // setCounter( counter -1)
        setCounter( (c) => c - 1)
    }
    // handleReset 
    const handleReset = (e) =>{
        
        setCounter(value)
    }

    // retornamos el jsx 
    return (
        <>
            <h1> CounterApp </h1>
            <p> { counter } </p>
            <button onClick={ handleSubtract }>-1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleAdd }>+1</button>
        </>
    );
}

// establecemos que value es de tipo numero y es requerido 
CounterApp.propTypes = {
    value: PropTypes.number
}

// exportamos el componente 
export default CounterApp;