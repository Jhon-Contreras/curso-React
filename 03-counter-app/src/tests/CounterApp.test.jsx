import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import CounterApp from "../CounterApp";
import { shallow } from 'enzyme';

describe('Pruebas en CounterApp.jsx', () => {
  
    test('Debe de mostrar <CounterApp /> correctamente (hacer match con un snapshot)', () => {
      
      
        const wrapper = shallow(<CounterApp/>);
        expect(wrapper).toMatchSnapshot();

    });
    test('debe de mostrar el valor por defecto de 100', () => {
       
        const wrapper = shallow(<CounterApp value={ 100 }/>);
        const counterText = wrapper.find('p').text().trim();
        expect(counterText).toBe('100');
       
    });
    
    
});
