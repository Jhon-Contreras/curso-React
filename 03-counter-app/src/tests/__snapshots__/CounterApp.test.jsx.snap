// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp.jsx Debe de mostrar <CounterApp /> correctamente (hacer match con un snapshot) 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <p>
     
    10
     
  </p>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;

exports[`Pruebas en CounterApp.jsx debe de mostrar el valor por defecto de 100 1`] = `"100"`;
